MultiSource/Benchmarks/Ptrdist/anagram/anagram.c

```diff
@@ -223,24 +226,32 @@ typedef struct {
 } Letter;
 typedef Letter * PLetter;
 
-Letter alPhrase[ALPHABET]; /* statistics on the current phrase */
+// CHECKEDC : checkedc array type
+// checkedc array type of T == array pointer type of T
+// bounds information for checkedc array is from array size
+Letter alPhrase _Checked [ALPHABET]; /* statistics on the current phrase */
 #define lPhrase(ch) alPhrase[ch2i(ch)]  /* quick access to a letter */
 
 int cchPhraseLength;                    /* number of letters in phrase */
 
-Quad aqMainMask[MAX_QUADS];/* the bit field for the full phrase */
-Quad aqMainSign[MAX_QUADS];/* where the sign bits are */
+Quad aqMainMask _Checked [MAX_QUADS];/* the bit field for the full phrase */
+Quad aqMainSign _Checked [MAX_QUADS];/* where the sign bits are */
+
+char achPhrase _Checked [255];
 
 int cchMinLength = 3;
 
 /* auGlobalFrequency counts the number of times each letter appears, summed
  * over all candidate words.  This is used to decide which letter to attack
  * first.
  */
-unsigned auGlobalFrequency[ALPHABET];
-char achByFrequency[ALPHABET];          /* for sorting */
+unsigned auGlobalFrequency _Checked [ALPHABET];
+char achByFrequency _Checked [ALPHABET];          /* for sorting */
 
-char * pchDictionary;               /* the dictionary is read here */
+// CHECKEDC : externally scoped variable bounds information
+// global variable bounds information is defined in declaration
+// dictionary maximum length is defined as MAXWORDS
+_Array_ptr<char> pchDictionary : count(MAXWORDS);               /* the dictionary is read here */
```

Comment:
Sorry, this spec is imprecise here.  It should read, "by definition their bounds declarations are always program invariants".    NULL satisfies any invariant, so pchDictionary can be initialized to 0, no matter what the value of pchDictionarySize.

Technically, we will need a `bundle` block around the initialization of pchDictionarySize and pchDictionary in `ReadDict`.  I will correct the spec.